analyzers/src/SonarAnalyzer.Common/Rules/StringLiteralShouldNotBeDuplicatedBase.cs

```diff
@@ -87,10 +87,17 @@ private void ReportOnViolation(SyntaxNodeAnalysisContext context)
             {
                 var duplicates = item.ToList();
                 var firstToken = duplicates[0];
+                var messageText = ExtractStringContent(firstToken);
                 context.ReportIssue(Diagnostic.Create(rule, firstToken.GetLocation(),
                     duplicates.Skip(1).Select(x => x.GetLocation()),
-                    item.Key, duplicates.Count));
+                    messageText, duplicates.Count));
             }
         }
+
+        private static string ExtractStringContent(SyntaxToken literalToken) =>
+             // Use literalToken.Text to get the text as written by the developer. The unescaped text (literalToken.ValueText)
+             // might contain control characters that may cause trouble when used as error message (e.g. a null-terminator).
+             // The literalToken.Text contains leading and trailing double quotes that we strip of.
+             literalToken.Text.StartsWith("@\"") ? literalToken.Text.Substring(2, literalToken.Text.Length - 3) : literalToken.Text.Substring(1, literalToken.Text.Length - 2);
```

Comment:
What happened with the original `StripOfLeadingAndTrialingDoubleQuotes` idea? It doesn't support `@"` scenario?

Reply:
I made that name up to illustrate what I was going to do. I don't think we already have something like that in the code base.

Reply:
Ah, ok 😄